hw4/04-templates.js

```diff
@@ -14,51 +14,77 @@ const url = 'https://restcountries.com/v3.1/all';
+    });
 });
 
-app.get('/capitals', (req, res) => {
-  // map the output array to create an array with country names and capitals
-  // check for empty data in the output array
-
-  let countries = ['Afghanistan', 'Aland Islands', 'Albania'];
-
-  res.render('page', {
-    heading: 'Countries and Capitals',
-    results: countries,
-  });
+app.get('/capitals', async (req, res) => {
+    // map the output array to create an array with country names and capitals
+    // check for empty data in the output array
+    getCountries((data) => {
+        let countryAndCapitals = data
+            .map((response) => {
+                return `${response.name.common ? response.name.common : 'Not found'} - ${
+                    response.capital ? response.capital : 'Not found'
+                }`;
+            })
+            .sort();
+
+        res.render('page', {
+            heading: 'Countries and Capitals',
+            results: countryAndCapitals,
+        });
+    }).catch((err) => console.log(err));
 });
 
 app.get('/populous', (req, res) => {
-  // filter the output array for the countries with population of 50 million or more
-  // sort the resulting array to show the results in order of population
-  // map the resulting array into a new array with the country name and formatted population
-
-  let populous = ['China', 'India', 'United States of America'];
-
-  res.render('page', {
-    heading: 'Most Populous Countries',
-    results: populous,
-  });
+    // filter the output array for the countries with population of 50 million or more
+    // sort the resulting array to show the results in order of population
+    // map the resulting array into a new array with the country name and formatted population
+    getCountries((data) => {
+        const result = data
+            .filter((response) => response.population > 50000000)
+            .map((response) => {
+                return `${response.name.common} - ${response.population}`;
+            })
+            .sort();
+        res.render('page', {
+            heading: 'Most Populous Countries',
+            results: result,
+        });
+    }).catch((err) => console.log(err));
 });
 
 app.get('/regions', (req, res) => {
-  // reduce the output array in a resulting object that will feature the numbers of countries in each region
-  // disregard empty data from the output array
-
-  let regions = ['Asia - 50', 'Europe - 53', 'Africa - 60'];
-
-  res.render('page', {
-    heading: 'Regions of the World',
-    results: regions,
-  });
+    // reduce the output array in a resulting object that will feature the numbers of countries in each region
+    // disregard empty data from the output array
+    getCountries((data) => {
+        const regionsData = data.reduce((accumulator, randomNumber) => {
+            accumulator[randomNumber.region] = ++accumulator[randomNumber.region] || 0;
+            return accumulator;
+        }, {});
```

Comment:
I really liked your solution for "capitals" and "populous", but this got me very confused. I'm not exactly sure what is going on with this method and I think the randomNumber variable might be the thing throwing me off. Everything seems to be displaying correctly though so whatever it's doing seems to be correct.